src/aegis_ai/features/__init__.py

```diff
@@ -35,22 +60,28 @@ async def run_if_safe(self, prompt, **kwargs):
                 logger.warning(msg)
                 raise RuntimeError(msg)
 
-            try:
-                runner = self.agent.run(prompt.to_string(), **kwargs)
-                result = await asyncio.wait_for(runner, timeout=llm_prompt_timeout)
+            # will be merged with self.agent.model_settings by pydantic_ai
+            model_settings = {}
 
-            except asyncio.TimeoutError:
-                msg = f"{call_str}: LLM request timed out after {llm_prompt_timeout} seconds"
-                logger.warning(msg)
-                raise RuntimeError(msg)
+            # retry loop
+            for attempt in range(agent_default_max_retries):
+                try:
+                    result = await self._run(
+                        call_str, prompt, model_settings=model_settings, **kwargs
+                    )
+                    break
+                except UnexpectedModelBehavior as e:
+                    if "RECITATION" not in str(e):
+                        # propagate other exceptions
+                        raise
 
-            except Exception as e:
-                # log only exception name by default, details only when debugging
-                logger.warning(
-                    f"{call_str} raised an exception: {e.__class__.__name__}"
-                )
-                logger.debug(f"{call_str} raised an exception: {e}")
-                raise
+                    # retry with high temperature
+                    # see https://github.com/RedHatProductSecurity/aegis-ai/issues/271
+                    model_settings["temperature"] = PROMPT_RETRY_TEMPERATURE
```

Comment:
this is good ... though it might be in the future we use a different agent for retries ... if we find we are 'tickling' more then just model settings.